torchio/transforms/preprocessing/intensity/projection.py

```diff
@@ -0,0 +1,158 @@
+            number of voxels in the ``axis`` dimension to project across.
+            If ``None``, the projection will be done across the entire span of
+            the ``axis`` dimension (i.e. ``axis`` dimension will be reduced to
+            1).
+        stride: Number of voxels to stride along the ``axis`` dimension between
+            slab projections.
+        projection_type: Type of intensity projection. Possible inputs are
+            ``'max'`` (the default), ``'min'``, ``'mean'``, ``'median'``, or
+            ``'quantile'``. If ``'quantile'`` is used, ``q`` must also be
+            supplied.
+        q: Quantile to use for intensity projections. This argument is required
+            if ``projection_type`` is ``'quantile'`` and is silently ignored
+            otherwise.
+        full_slabs_only: Boolean. Should projections be done only for slabs
+            that are ``slab_thickness`` thick? Default is ``True``.
+            If ``False``, some slabs may not be ``slab_thickness`` thick
+            depending on the size of the image, slab thickness, and stride.
+
+    Example:
+        >>> import torchio as tio
+        >>> ct = tio.datasets.Slicer('CTChest').CT_chest
+        >>> axial_mips = tio.Projection("S", slab_thickness=20, stride=20)
+        >>> ct_t = axial_mips(ct)
+        >>> ct_t.plot()
+
+    .. plot::
+
+        import torchio as tio
+        sub = tio.datasets.Slicer('CTChest')
+        ct = sub.CT_chest
+        axial_mips = tio.Projection("S", slab_thickness=20, stride=20)
+        ct_mips = axial_mips(ct)
+        sub.add_image(ct_mips, 'MIP')
+        sub.plot()
+
+    """
+    def __init__(
+            self,
+            axis: str,
+            slab_thickness: Optional[int] = None,
+            stride: Optional[int] = 1,
+            projection_type: Optional[str] = 'max',
+            q: Optional[float] = None,
+            full_slabs_only: Optional[bool] = True,
+            **kwargs
+            ):
+        super().__init__(**kwargs)
+        self.args_names = (
+            'axis', 'slab_thickness', 'stride',
+            'projection_type', 'q', 'full_slabs_only'
+            )
+        self.axis = axis
+        self.slab_thickness = slab_thickness
+        self.stride = stride
+        self.projection_type = projection_type
+        self.q = q
+        self.full_slabs_only = full_slabs_only
+        self.projection_fun = self.get_projection_function()
+
+    def get_projection_function(self):
+        if self.projection_type == 'max':
+            projection_fun = torch.amax
+        elif self.projection_type == 'min':
+            projection_fun = torch.amin
+        elif self.projection_type == 'mean':
+            projection_fun = torch.mean
+        elif self.projection_type == 'median':
+            projection_fun = torch.median
+        elif self.projection_type == 'quantile':
+            projection_fun = torch.quantile
+            self.validate_quantile()
```

Comment:
Would it make sense to move this to `__init__`? So errors are caught as early as possible.

Reply:
`self.get_projection_function` gets called in `__init__` so `self.validate_quantile` does get called down the stack, but only if `projection_type='quantile'`. I did this so that `q` could be ignored but I could also just change `validate_quantile` to `pass` if `projection_type` is not `'quantile'` and it would be more obvious that it would throw errors during init.

Reply:
Makes sense. But I think the validation is cheap enough that it can be computed at instantiation time, to avoid frustrating runtime errors (e.g. once training has started).